### ISSUE TITLE
Allow renames in autoautosummary

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo-sphinx-theme#47]
Something that's come up several times in Nengo core is that we can't use `autoautosummary` in situations where we want the user to use an object in a module that it's not originally defined in. For example, we want users to do something like `nengo.Ensemble(neuron_type=nengo.LIF())` even though `Ensemble` is actually `nengo.ensemble.Ensemble` and `LIF` if `nengo.neurons.LIF`.

It would be nice to have a way to still use `autoautosummary` (so new objects aren't missed) but still be able to ensure that certain objects are referred to by our preferred name.

**Comment by tbekolay** [comment]
Actually, a viable alternative to figuring out a way to do renames would be to simply make manually specified entries go first in the list, as this is what the `automodule` directive does. It would mean that you can convert something like:

```rst
.. automodule:: nengo.builder.operator

   .. autoautosummary:: nengo.builder.operator
```

to this, renaming `Operator` to `nengo.builder.Operator`.

```rst
.. automodule:: nengo.builder.operator
   :exclude-member: Operator

   .. autoautosummary:: nengo.builder.operator
      :exclude-member: Operator

      nengo.builder.Operator

   .. autoclass:: nengo.builder.Operator
```

As a nice bonus, the things that we pull into higher namespaces happen to be the things that we generally want to emphasize (and thus put first) anyway.

**Comment by hunse** [comment]
Fixed in #52.
